src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java

```diff
@@ -36,130 +32,38 @@
-    private int currentCycle = Integer.MIN_VALUE;
-    private SingleChronicleQueueStore currentCycleWireStore;
-    private MappedBytes currentCycleMappedBytes;
+    static {
+        SingleChronicleQueueBuilder.onlyAvailableInEnterprise("Pretoucher");
+    }
 
+    @Deprecated(/* remove in x.25 */)
     public Pretoucher(final SingleChronicleQueue queue) {
-        this(queue,
-                null,
-                c -> {
-                },
-                Jvm.getBoolean("SingleChronicleQueueExcerpts.earlyAcquireNextCycle", false),
-                !Jvm.getBoolean("SingleChronicleQueueExcerpts.dontWrite"));
     }
 
-    // visible for testing
+    @Deprecated(/* remove in x.25 */)
     public Pretoucher(final SingleChronicleQueue queue,
                       final NewChunkListener chunkListener,
                       final IntConsumer cycleChangedListener,
                       boolean earlyAcquireNextCycle,
                       boolean canWrite) {
-        this.queue = queue;
-        this.chunkListener = chunkListener;
-        this.cycleChangedListener = cycleChangedListener;
-        this.earlyAcquireNextCycle = checkEA(earlyAcquireNextCycle);
-        this.canWrite = canWrite || this.earlyAcquireNextCycle;
-        pretoucherState = new PretoucherState(this::getStoreWritePosition);
-        if (PRETOUCHER_PREROLL_TIME_MS != PRETOUCHER_PREROLL_TIME_DEFAULT_MS && !earlyAcquireNextCycle)
-            Jvm.warn().on(getClass(), "SingleChronicleQueueExcerpts.pretoucherPrerollTimeMs has been set but not earlyAcquireNextCycle");
-        pretouchTimeProvider = () -> queue.time().currentTimeMillis() + (this.earlyAcquireNextCycle ? PRETOUCHER_PREROLL_TIME_MS : 0);
-
-        if (! SingleChronicleQueueBuilder.isQueueReplicationAvailable())
-            Jvm.warn().on(getClass(), "This functionality has been deprecated and in future will only be available in Chronicle Queue Enterprise");
-        // always put references to "this" last.
-        queue.addCloseListener(this);
-    }
-
-    private boolean checkEA(boolean ea) {
-        if (ea)
-            Jvm.warn().on(getClass(), "SingleChronicleQueueExcerpts.earlyAcquireNextCycle is not supported");
-        return false;
     }
 
     public void execute() throws InvalidEventHandlerException {
-        try {
-            throwExceptionIfClosed();
-        } catch (ClosedIllegalStateException e) {
-            throw new InvalidEventHandlerException(e);
-        }
-
-        try {
-            assignCurrentCycle();
-
-            if (currentCycleMappedBytes != null)
-                pretoucherState.pretouch(currentCycleMappedBytes);
-
-        } catch (ClassCastException cce) {
-            Jvm.warn().on(getClass(), cce);
-
-        } catch (IllegalStateException e) {
-            if (queue.isClosed())
-                throw new InvalidEventHandlerException(e);
-            else
-                Jvm.warn().on(getClass(), e);
-        }
+        // do nothing
```

Comment:
I think it should at least produce a warning if used.

Reply:
It will - see static block in Pretoucher